Source/Details/ASRangeController.mm

```diff
@@ -376,7 +380,7 @@ - (void)_updateVisibleNodeIndexPaths
         [newVisibleNodes addObject:node];
       }
       // Skip the many method calls of the recursive operation if the top level cell node already has the right interfaceState.
-      if (node.interfaceState != interfaceState) {
+      if (node.pendingInterfaceState != interfaceState) {
```

Comment:
This will make sure the correct interface exiting order when thrashing happen during push animation (node being pushed will exit - enter - exit hierarchy).  
Note the first exit & enter happened in one runloop triggered by CAAnimation inside UIKit
before the change: node will exit for first exit call instead of last.